feat: bottomNavigation 기본 세팅

s9hn opened the pull request:
## 📌𝘐𝘴𝘴𝘶𝘦𝘴
- closed #2 

## 📎𝘞𝘰𝘳𝘬 𝘋𝘦𝘴𝘤𝘳𝘪𝘱𝘵𝘪𝘰𝘯
- 기본 바텀네비게이션 세팅 및 프레임 프래그먼트 디렉터리 구조 잡아놓았습니다.
- 추후 변경될 수 있어 기본형으로 세팅했습니다.

## 📷𝘚𝘤𝘳𝘦𝘦𝘯𝘴𝘩𝘰𝘵


## 💬𝘛𝘰 𝘙𝘦𝘷𝘪𝘦𝘸𝘦𝘳𝘴

librarywon commented on app/src/main/res/values/typography.xml:
```diff
@@ -0,0 +1,50 @@
+    <style name="title1">
+        <item name="fontFamily">@font/pretendard_medium</item>
+        <item name="android:textSize">18sp</item>
+        <item name="lineHeight">25.2sp</item>
```
소수점 단위 sp 제거해주세요!

librarywon commented on app/src/main/res/values/colors.xml:
```diff
@@ -2,4 +2,16 @@
 <resources>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
+    <color name="primary_50">#F1EFFF</color>
```
```suggestion
    <color name="primary_50_F1EFFF">#F1EFFF</color>
```
이런 형식의 컬러 네이밍 컨벤션을 설정하였습니다.

librarywon reviewed (approved):
고생했습니다. 다른 사람 어프로브 나면 머지하십시오!

m6z1 reviewed (approved):
굿

junseo511 reviewed (approved):
굿 👍

s9hn committed:
feat: base class, 프레임 프래그먼트 생성 및 적용

s9hn committed:
feat: 바텀 네비게이션 세팅

- 바텀 네비게이션 기본형 세팅 및 적용하였습니다.
- 추후 상세 디자인이 바뀔 수 있어 기본 세팅만 적용하였습니다.
- 현재는 replace 방식이 적용되어 있습니다. 추후 변경될 수 있습니다.

m6z1 commented on app/src/main/java/com/teamwss/websoso/ui/common/base/BindingFragment.kt:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
```
저도 코드리뷰 받은 부분인데 binding 을 null값으로 해제해주는 걸 뷰를 `destroy` 하기 전에 먼저 한대요
그 이유는 메모리의 누수가 있을 수도 있다 ?

s9hn replied:
과연 그럴까요?

librarywon replied:
<img width="719" alt="스크린샷 2024-03-21 오전 12 07 04" src="https://github.com/Team-WSS/WSS-Android/assets/52442547/4f6a7c7e-a81e-4d6e-9851-4769767a2b01">
공식문서도 이 순서를 따르긴하네요

https://developer.android.com/topic/libraries/view-binding

junseo511 replied:
저도 위와 같이 알고있어요!!

m6z1 commented on app/src/main/java/com/teamwss/websoso/ui/main/MainActivity.kt:
```diff
@@ -1,17 +1,66 @@
+        return true
+    }
+
+    private inline fun <reified T : Fragment> replaceFragment() {
```
`reified` 가 무엇인가여 ?

s9hn replied:
공부해서 팀원들에게 공유해주실 수 있나요?

librarywon replied:
공부해야겠네...

junseo511 replied:
야호 명지의 미미나다

m6z1 commented on app/src/main/java/com/teamwss/websoso/ui/main/MainActivity.kt:
```diff
@@ -1,17 +1,66 @@
+        }
+    }
+
+    private enum class FragmentType(@IntegerRes private val resId: Int) {
+        HOME(R.id.menu_home),
+        EXPLORE(R.id.menu_explore),
+        FEED(R.id.menu_feed),
+        MY_PAGE(R.id.menu_my_page)
+        ;
```
FragmentType을 `private enum class` 로 선언하신 이유는 메인 액티비티에서만 사용되는 것이고 enum class 로 선언하려고 해서인가요 ? ? 액티비티에 저렇게 선언하신 이유가 궁금합니다.

s9hn replied:
resId값을 이넘으로 포장해 가독성을 높이며 메인에서만 쓰기에 프라이빗으로 뒀습니다

m6z1 reviewed (approved):
Apple TM  ~

librarywon commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,13 @@
+    <string name="base_fragment_null_error">bindingException: binding is null in BindingFragment</string>
+
+    <!-- 메인 뷰 -->
+    <string name="main_ic_home">홈</string>
+    <string name="main_ic_explore">탐색</string>
+    <string name="main_ic_feed">피드</string>
+    <string name="main_ic_my_page">my</string>
```
엑티비티_용도_설명인가요?

s9hn replied:
PRND 스트링 컨벤션 참고했습니다.
`<WHERE>_<DESCRIPTION>`

librarywon replied:
확인하였습니다!

librarywon reviewed (approved):
Base Activity, Fragment 파일 이름 앞에 Base를 붙이시지 않은 이유가 있나요?

s9hn commented:
> Base Activity, Fragment 파일 이름 앞에 Base를 붙이시지 않은 이유가 있나요?

BaseClass is Evil 이라는 말 아시나요?

모든 뷰에서 베이스클래스를 참조하고 있다면, 베이스클래스에 대해 강한 결합을 하고 있다는 의미니 조심해야겠죠

하나의 자원 혹은 클래스가 여러 곳에서 사용되는 경우는 항상 주의해야합니다 ! 
(싱글톤이든 확장함수든 바인딩어댑터든 말이죠)

같은 이유로 저도 베이스 클래스를 선호하지는 않습니다만, 바인딩 정도는 충분히 보일러 플레이트 코드를 줄이는데 이점이 있다고 생각하고 별 다른 사이드 이펙트를 발생시키지 않는다고 생각해서 바인딩 목적으로만 사용합니다.

그래서 이름을 그냥 BaseClass가 아닌 바인딩 클래스로 두었씁니다

librarywon commented:
Base클레스관련한 코멘트 확인하였습니다.` BaseClass is Evil `은 처음 들었던 문구네요 항상 배워갑니다!

librarywon commented on app/src/main/java/com/teamwss/websoso/ui/main/MainActivity.kt:
```diff
@@ -1,17 +1,66 @@
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.commit
+import androidx.fragment.app.replace
+import com.created.team201.presentation.common.BindingActivity
```
이 의존성은 뭔가요?

junseo511 reviewed (approved):
어푸합니다~~!! 피알을 이제 발견했지 무얌 머쓱 ;; 🤣 리뷰어 등록해주세용

s9hn committed:
refactor: 린트 체크 및 패키지 수정

yeonjeen commented on app/src/main/java/com/teamwss/websoso/ui/main/MainActivity.kt:
```diff
@@ -1,17 +1,66 @@
+            EXPLORE -> replaceFragment<ExploreFragment>()
+            FEED -> replaceFragment<FeedFragment>()
+            MY_PAGE -> replaceFragment<MyPageFragment>()
+        }
```
여기도 null 처리 해줘야하지 않나용?

yeonjeen reviewed (approved):
GSTM
